src/cdk/v2/destinations/ortto/procWorkflow.yaml

```diff
@@ -0,0 +1,140 @@
+      $.assert(messageType in {{$.EventType.([.IDENTIFY, .TRACK])}}, 
+        "message type " + messageType + " is not supported")
+      console.log($.getGenericPaths("email"));
+      $.assert(.message.().({{{{$.getGenericPaths("email")}}}}) || .message.().({{{{$.getGenericPaths("userId")}}}}), "email  or userId is a required field. Aborting message.");
+
+  - name: messageType
+    template: |
+      .message.type.toLowerCase()
+
+  - name: validateInputForTrack
+    description: Additional validation for Track events
+    condition: $.outputs.messageType === {{$.EventType.TRACK}}
+    template: |
+      $.assert(.message.event, "event is not present. Aborting.")
+
+  - name: commonFields
+    description: |
+      Builds common fields in destination payload.
+    template: |
+      let commonFields = .message.().({
+        "fields": { 
+          "str::first": {{{{$.getGenericPaths("firstName")}}}}, 
+          "str::last": {{{{$.getGenericPaths("lastName")}}}},
+          "str::email": {{{{$.getGenericPaths("email")}}}},  
+          "geo::city": {"name":{{{{$.getGenericPaths("city")}}}}},
+          "geo::country": {"name":{{{{$.getGenericPaths("country")}}}}},
+          "geo::region": {"name":{{{{$.getGenericPaths("region")}}}}},
+          "str::postal": {{{{$.getGenericPaths("zipcode")}}}},
+          "str::ei": {{{{$.getGenericPaths("userId")}}}},
+          "str::language": .message.context.traits.language || .message.context.locale,
+          "phn::phone": {"n": {{{{$.getGenericPaths("userId")}}}}},
+          "bol::gdpr": true,
+          "bol::p": false,
+          "bol::sp": false,
+          },
+        "location": {"source_ip": .message.context.ip}
+      });
+      .destination.Config.orttoPersonAttributes@attribute.(
+              commonFields.fields[$.fieldTypeMap[attribute.type]+":cm:"+attribute.orttoAttribute.trim().toLowerCase().replace(new RegExp('\\s+', 'g'),'-')] = $.originalInput.message.context.traits[attribute.rudderTraits]
+            )[]
+      $.removeUndefinedAndNullValues(commonFields)
+
+  - name: validateCommonFields
+    template: |
+      const external_id = $.outputs.commonFields.fields."str::ei";
+      const email = $.outputs.commonFields.fields."str::email";
+      $.assert(external_id || email, "userId or email is required");
+
+  - name: prepareIdentifyPayload
+    condition: $.outputs.messageType === {{$.EventType.IDENTIFY}}
+    steps:
+      - name: propertiesFields
+        template: |
+          const peopleObj = {
+            "fields": $.outputs.commonFields.fields,
+            "tags": .message.context.traits.tags || .message.traits.tags || .message.prooperties.tags,
+            "unset_tags": .message.context.traits.unset_tags || .message.traits.unset_tags || .message.prooperties.unset_tags
+          }
+          const identifyPayoad = {                                  
+            "people":[peopleObj],
+            "merge_by": ["external_id", "email"]
```

Comment:
What if email is not available in the first place?

Reply:
It first checks with `external_id` and if not available then `email`, if it finds the profile, it will be updated else created.